File: webapp-java/dxa-module-ugc/src/main/java/com/sdl/dxa/modules/ugc/data/PostedComment.java
package com.sdl.dxa.modules.ugc.data;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

/**
 * <p>Ugc posted comment</p>
 */

@Data
public class PostedComment {
    @JsonProperty("publicationId")
    private int publicationId;
    @JsonProperty("pageId")
    private int pageId;
    @JsonProperty("username")
    private String userName;
    @JsonProperty("email")
    private String email;
    @JsonProperty("content")
    private String content;
    @JsonProperty("parentId")
    private int parentId = 0;
    @JsonProperty("publicationTitle")
    private String publicationTitle;
    @JsonProperty("formUrl")
    private String formUrl;
    @JsonProperty("target")
    private String target;
    @JsonProperty("publicationUrl")
    private String publicationUrl;
    @JsonProperty("pageTitle")
    private String pageTitle;
    @JsonProperty("pageUrl")
    private String pageUrl;
    @JsonProperty("language")
    private String language;
    @JsonProperty("status")
    private String status;
    @JsonProperty("itemTitle")
    private  String itemTitle;
    @JsonProperty("itemUrl")
    private String itemUrl;
}

File: webapp-net/Ugc/Data/PostedComment.cs
namespace Sdl.Web.Modules.Ugc.Data
{
    /// <summary>
    /// Posted Comment
    /// </summary>
    public class PostedComment
    {
        public int PublicationId { get; set; }

        public int? PageId { get; set; }

        public string Username { get; set; }

        public string Email { get; set; }

        public string Content { get; set; }

        public int? ParentId { get; set; } = 0;

        public string PublicationTitle { get; set; }

        public string PublicationUrl { get; set; }

        public string PageTitle { get; set; }

        public string PageUrl { get; set; }

        public string Language { get; set; }
    }
}
